relm4-components/src/simple_adw_combo_row.rs

```diff
@@ -0,0 +1,121 @@
+
+use adw::gtk::StringList;
+use adw::prelude::ComboRowExt;
+use relm4::ComponentSender;
+use relm4::{Component, ComponentParts};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// A simple wrapper around [`adw::ComboRow`].
+///
+/// This can be used with enums, [`String`]s or any custom type you want.
+/// The only requirement is that the inner type implements [`ToString`] and [`Debug`].
+///
+/// To get notified when the selection changed, you can use
+/// [`Connector::forward()`](relm4::component::Connector::forward())
+/// after launching the component.
+pub struct SimpleComboRow<E: ToString> {
+    /// The variants that can be selected.
+    pub variants: Vec<E>,
+    /// The index of the active element or [`None`] is nothing is selected.
+    pub active_index: Option<usize>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// The message type of [`SimpleComboRow`].
+pub enum SimpleComboRowMsg<E: ToString> {
+    /// Overwrite the current values.
+    UpdateData(SimpleComboRow<E>),
+    /// Set the index of the active element.
+    SetActiveIdx(usize),
+    #[doc(hidden)]
+    UpdateIndex(usize),
+}
+
+impl<E> Component for SimpleComboRow<E>
+where
+    E: ToString + 'static + Debug,
+{
+    type CommandOutput = ();
+    type Input = SimpleComboRowMsg<E>;
+    type Output = usize;
+    type Init = Self;
+    type Root = adw::ComboRow;
+    type Widgets = adw::ComboRow;
+
+    fn init_root() -> Self::Root {
+        adw::ComboRow::default()
+    }
+
+    fn init(
+        model: Self::Init,
+        root: &Self::Root,
+        sender: ComponentSender<Self>,
+    ) -> ComponentParts<Self> {
+        let widgets = root.clone();
+
+        model.render(&widgets);
+
+        widgets.connect_selected_notify(move |combo_box| {
+            sender.input(Self::Input::UpdateIndex(combo_box.selected() as _));
+        });
+
+        ComponentParts { model, widgets }
+    }
+
+    fn update_with_view(
+        &mut self,
+        widgets: &mut Self::Widgets,
+        input: Self::Input,
+        sender: ComponentSender<Self>,
+        _root: &Self::Root,
+    ) {
+        match input {
+            SimpleComboRowMsg::UpdateIndex(idx) => {
+                // Ignore send errors because the component might
+                // be detached.
+                sender.output(idx).ok();
+                self.active_index = Some(idx);
+            }
+            SimpleComboRowMsg::SetActiveIdx(idx) => {
+                if idx < self.variants.len() {
+                    self.active_index = Some(idx);
+                    widgets.set_selected(idx as u32);
+                }
+            }
+            SimpleComboRowMsg::UpdateData(data) => {
+                *self = data;
+                self.render(widgets);
+            }
+        }
+    }
+}
+
+impl<E> SimpleComboRow<E>
+where
+    E: ToString,
+{
+    fn render(&self, combo_box: &adw::ComboRow) {
+        use std::ops::Deref;
+        if combo_box.model().is_some() {
+            unreachable!()
+        }
```

Comment:
This isn't exactly unreachable. It is possible for the user to add entries by accessing the combo row as root widget manually. And I think the `UpdateData` message would also trigger a panic here.
The question is whether this needs to be checked at all. If the user does stuff manually, they should know what they do and expect unexpected behavior.

Reply:
Doesn't the `required-features` field make this work for `--all-targets`?

Reply:
Good catch!

I removed the check.